spec/enumerables_spec.rb

```diff
@@ -0,0 +1,274 @@
+require './enumerable.rb'
+
+describe Enumerable do
+  let(:array_integers) { (1..10).to_a }
+  let(:array_integers_false) { (1..9).to_a }
+  let(:array_strings) { %w[John Mary Douglas Michael] }
+  let(:array_strings_false) { %w[John Mary Douglas] }
+  let(:array_reduced) { 3_628_800 }
+  let(:true_array) { [1, true, 'hi', []] }
+  let(:false_array) { [1, false, 'hi', []] }
+  let(:false_array_2) { [false, 'hi', []] }
+  let(:range_string) { ('a'..'b') }
+  let(:range_integers) { (1..10) }
+  let(:hash) { [{ 'Michael' => 2 }, { 'John' => 3 }] }
+  let(:result) { [] }
+  let(:compare) { [] }
+
+  describe '#my_each' do
+    it 'Return every element of an array of integers passing a block' do
+      array_integers.my_each { |n| result << n + 1 }
+      array_integers.each { |n| compare << n + 1 }
+      expect(result).to eq(compare)
+    end
+
+    it 'Return every element of an array of strings' do
+      array_strings.my_each { |n| result << n }
+      array_strings.each { |n| compare << n }
+      expect(result).to eq(compare)
+    end
+
+    it 'Return every element of a Hash' do
+      hash.my_each { |n| result << n }
+      hash.each { |n| compare << n }
+      expect(result).to eq(compare)
+    end
+  end
+
+  describe '#my_each_with_index' do
+    it 'Return every element of an array of integers passing a block' do
+      array_integers.my_each_with_index { |n| result << n + 1 }
+      array_integers.each { |n| compare << n + 1 }
+      expect(result).to eq(compare)
+    end
+
+    it 'Return every element of an array of strings' do
+      array_strings.my_each_with_index { |n| result << n }
+      array_strings.each { |n| compare << n }
+      expect(result).to eq(compare)
+    end
+
+    it 'Return every element of a Hash' do
+      hash.my_each_with_index { |_key, value| result << value + 2 }
+      hash.each_with_index { |_key, value| compare << value + 2 }
+      expect(result).to eq(compare)
+    end
+  end
+
+  describe '#my_select' do
+    it 'Checks if a element is selected from array' do
+      result = array_integers.my_select { |n| n > 9 }
+      compare = [10]
+      expect(result).to eq(compare)
+    end
+
+    it 'Checks if a element is selected from array of strings' do
+      result = array_strings.my_select { |n| n == 'John' }
+      compare = ['John']
+      expect(result).to eq(compare)
+    end
+
+    it 'Checks if a element is selected from a hash' do
+      result = hash.my_select { |n| n['John'] }
+      compare = hash.select { |n| n['John'] }
+      expect(result).to eq(compare)
+    end
+  end
+
+  describe '#my_all?' do
+    it 'Checks for true if all elements in a integer array matchs all the conditions' do
+      result = array_integers.my_all?(array_integers.all?)
```

Comment:
You are calling the main enumerable method inside yours as an argument isn't proper.
 `array_integers.my_all?(array_integers.all?)`
- [ ] You could check for when a block is given.



**Note:** Lets apply this to all `it` block in this scenario